lib/WebSocket.js

```diff
@@ -808,35 +807,16 @@ function establishConnection(ReceiverClass, SenderClass, socket, upgradeHead) {
   ultron.on('error', cleanupWebsocketResources.bind(this));
 
   // ensure that the upgradeHead is added to the receiver
-  function firstHandler(data) {
-    if (called || self.readyState === WebSocket.CLOSED) return;
```

Comment:
The `data` event is not emitted on the same tick but I guess the `data` listener is removed when `WebSocket` is closed so it should be fine.